packages/core/src/oidc/grants/token-exchange/types.ts

```diff
@@ -0,0 +1,9 @@
+import { z } from 'zod';
+
+export const tokenExchangeActGuard = z.object({
+  act: z.object({
+    sub: z.string(),
+  }),
+});
+
+export type TokenExchangeAct = z.infer<typeof tokenExchangeActGuard>;
```

Comment:
nit: define type and use `satisfies ToZodObject<TokenExchangeAct>` for the guard